src/main/java/org/junit/experimental/theories/internal/Assignments.java

```diff
@@ -85,11 +85,14 @@ public Assignments assignNext(PotentialAssignment source) {
 
     private List<PotentialAssignment> generateAssignmentsFromTypeAlone(ParameterSignature unassigned) {
         Class<?> paramType = unassigned.getType();
-        
-        if (paramType.isEnum()) {
-            return new EnumSupplier(paramType).getValueSources(unassigned);  
-        } else if (paramType.equals(Boolean.class) || paramType.equals(boolean.class)) {
-            return new BooleanSupplier().getValueSources(unassigned);
+
+        FromDataPoints fromDataPoints = unassigned.getAnnotation(FromDataPoints.class);
```

Comment:
❓ Shouldn't this check for `ParametersSuppliedBy` or, alternatively, shouldn't `SpecificDataPointsSupplier` throw an exception if it couldn't find a matching field?

Reply:
I think so